app/src/main/java/com/cornellappdev/transit/ui/viewmodels/FavoritesViewModel.kt

```diff
@@ -58,15 +59,30 @@ class FavoritesViewModel @Inject constructor(
     }.stateIn(scope, SharingStarted.Eagerly, emptyList())
 
 
+
     /**
-     * Function to remove a stop from favorites
+     * Asynchronous function to remove a stop from favorites
      */
-    suspend fun removeFavorite(stop: String?) {
+    fun removeFavorite(stop: String?) {
         if (stop != null) {
             val currentFavorites = favoritesFlow.value.toMutableSet()
             currentFavorites.remove(stop)
-            userPreferenceRepository.setFavorites(currentFavorites.toSet())
+            CoroutineScope(Dispatchers.IO).launch {
+                userPreferenceRepository.setFavorites(currentFavorites.toSet())
+            }
```

Comment:
This is in a viewmodel, so the best practice is to use `viewmodelScope.launch`, I think it's called.